### ISSUE TITLE
`non-local exit from iterator without return value` has a false positive with #define_method.

### ISSUE THREAD

**Issue by dnl** [bbatsov/rubocop#2272]
``` ruby
[:method_one, :method_two].each do |method_name|
  define_method(method_name) do
    return if predicate?
  end
end
```

is flagged with: 
```
Non-local exit from iterator, without return value. next, break, Array#find, Array#any?, etc. is preferred.
```

```
rubocop -V
0.34.2 (using Parser 2.2.2.6, running on ruby 2.2.2 x86_64-darwin14)
(and current master)
```

**Comment by alexdowad** [comment]
Just opening a PR which fixes this.
